tests/main.tftest.hcl

```diff
@@ -1,104 +1,96 @@
-variables {
-  vpc_id = "vpc-12345678"
-  subnet_ids = ["subnet-12345678", "subnet-87654321"]
-  stage = "test"
-  namespace = "mp"
-  name = "ssm-agent"
-  region = "us-east-1"
-  availability_zones = ["us-east-1a"]
-  nat_gateway_enabled = true
-  ipv6_enabled = true
-}
+### Integration Tests for the SSM Agent Module
+### This test suite will create the SSM Agent module
+### and validate the resources created by the module,
+### then destroy it.
+
+### `/test-harness/` module is used as a helper to validate resources that aren't in the Terraform state, for example the EC2 instances created from the ASG.
 
-### TESTING INSTANCE and ARCHITECTURE COMPATIBILITY ###
-# https://docs.aws.amazon.com/ec2/latest/instancetypes/instance-type-names.html
-# https://aws.amazon.com/ec2/instance-types/
+run "setup" {
+  module {
+    source = "./tests/setup"
+  }
+}
 
-# Test valid x86_64 instance type
-run "valid_x86_64_instance" {
-  command = plan
+run "create_ssm_agent" {
+  command = apply
 
   variables {
-    instance_type = "t3.micro"
-    architecture = "x86_64"
+    namespace = "mp"
+    stage     = "terraform-test${run.setup.random_number}"
   }
 
-  assert {
-    condition     = local.is_instance_compatible
-    error_message = "Expected instance type t3.micro to be compatible with x86_64 architecture"
+  module {
+    source = "./examples/complete"
   }
-}
 
-# Test valid arm64 instance type
-run "valid_arm64_instance" {
-  command = plan
-
-  variables {
-    instance_type = "t4g.micro"
-    architecture = "arm64"
+  assert {
+    condition     = module.ssm_agent.security_group_id != ""
+    error_message = "The ID of the SSM Agent Security Group is empty, possibly not created."
   }
 
   assert {
-    condition     = local.is_instance_compatible
-    error_message = "Expected instance type t4g.micro to be compatible with arm64 architecture"
+    condition     = module.ssm_agent.launch_template_id != ""
+    error_message = "The ID of the SSM Agent Launch Template is empty, possibly not created."
   }
-}
 
-# Test invalid x86_64 instance type (using arm64 instance type)
-run "invalid_x86_64_instance" {
-  command = plan
+  assert {
+    condition     = module.ssm_agent.autoscaling_group_id != ""
+    error_message = "The ID of the SSM Agent Autoscaling Group is empty, possibly not created."
+  }
```

Comment:
I question whether we should test this sort of thing... If the ASG didn't have an ID then it would have failed during the apply, correct? 

I feel like we should only be testing that X === Y. e.g. We might add `-asg` onto the label that we provide to the ASG module / resource... then we can test that the ASG name equals what we expect. 

If we do tests like this, we could just be creating unnecessary noise. 

This is my first reaction at least.